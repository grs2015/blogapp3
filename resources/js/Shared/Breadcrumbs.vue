<script setup lang="ts">
import { Link } from '@inertiajs/inertia-vue3';
import { ref } from 'vue';
import { usePage } from '@inertiajs/inertia-vue3';

interface breadcrumbsData {
    data: Array<{
        label: string,
        icon: string,
        route?: string
    }>

}

const props = defineProps<breadcrumbsData>()
const link = ref('/admin/posts')

</script>

<template>
    <Link as="div" :href="link">
        <q-breadcrumbs class="text-orange">
            <template #separator>
                <q-icon size="1.5em" name="chevron_right" color="primary" />
            </template>
            <template v-for="item in data" :key="item.label">
                <q-breadcrumbs-el :label="$t(item.label)" :icon="item.icon" :href="item.route" @click="link = item.route" />
            </template>
        </q-breadcrumbs>
    </Link>
</template>
