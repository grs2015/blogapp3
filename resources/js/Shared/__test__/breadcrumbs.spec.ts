import { mount } from "@vue/test-utils";
import { i18nVue } from "laravel-vue-i18n";
import { Quasar } from "quasar";
import Breadcrumbs from "@/Shared/Breadcrumbs.vue";

const breadcrumbsData = [
    {
        label: "Dashboard",
        icon: "home",
        route: "/admin",
    },
    {
        label: "Posts",
        icon: "widgets",
    },
]

describe("Breadcrumbs Component", () => {
    it('checks rendered text/links at component', () => {
        const wrapper = mount(Breadcrumbs, {
            global: {
                plugins: [i18nVue, Quasar],
            },
            props: { data: breadcrumbsData },
            shallow: false,
        });

        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.html()).toContain("Dashboard");
        expect(wrapper.html()).toContain("Posts");

        expect(wrapper.find('a').attributes('href')).toBe('/admin')
        expect(wrapper.find('a').text()).toContain('Dashboard')
    })
})
