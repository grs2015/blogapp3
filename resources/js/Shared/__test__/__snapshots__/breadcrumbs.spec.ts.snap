// Vitest Snapshot v1

exports[`Breadcrumbs Component > checks rendered text/links at component 1`] = `
<div>
  <div
    class="q-breadcrumbs text-orange"
  >
    <div
      class="flex items-center justify-start q-gutter-sm"
    >
      <div
        class="flex items-center text-primary"
      >
        <a
          class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable"
          href="/admin"
        >
          <i
            aria-hidden="true"
            class="q-icon notranslate material-icons q-breadcrumbs__el-icon q-breadcrumbs__el-icon--with-label"
            role="presentation"
          >
            home
          </i>
          Dashboard
        </a>
      </div>
      <div
        class="q-breadcrumbs__separator"
      >
        <i
          aria-hidden="true"
          class="q-icon text-primary notranslate material-icons"
          role="presentation"
          style="font-size: 1.5em;"
        >
          chevron_right
        </i>
      </div>
      <div
        class="flex items-center q-breadcrumbs--last"
      >
        <span
          class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable"
        >
          <i
            aria-hidden="true"
            class="q-icon notranslate material-icons q-breadcrumbs__el-icon q-breadcrumbs__el-icon--with-label"
            role="presentation"
          >
            widgets
          </i>
          Posts
        </span>
      </div>
    </div>
  </div>
</div>
`;
